src/yunohost/tools.py

```diff
@@ -555,3 +555,63 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+
+def tools_urlavailable(auth, url):
+    """
+    Check availability of a web path
+
+    Keyword argument:
+        url -- Url to check
+    """
+
+    domain, path = _parse_app_url(url)
+
+    # Abort if domain is unknown
+    if domain not in domain_list(auth)['domains']:
+        raise MoulinetteError(errno.EINVAL, m18n.n('domain_unknown'))
+
+    # Fetch apps map
+    apps_map = app_map(raw=True)
+
+    # Loop through all apps to check if path is taken by one of them
+    available = "Yes"
+    if domain in apps_map:
+        # Loop through apps
+        for p, a in apps_map[domain].items():
+            if path == p:
+                available = "No"
+            # We also don't want conflicts with other apps starting with
+            # same name
+            elif path.startswith(p) or p.startswith(path):
+                available = "No"
+
+
+    return { "available" : available }
+
+def _parse_app_url(url):
+
+    # We want url to be of the format :
+    #  some.domain.tld/foo
+
+    # Remove http/https prefix if it's there
+    if url.startswith("https://"):
+        url = url[len("https://"):]
+    elif url.startswith("http://"):
+        url = url[len("http://"):]
+
+    # Remove trailing slash
+    url = url.rstrip("/")
+
+    # Check url contains at least a slash
+    if "/" not in url:
+        raise MoulinetteError(errno.EINVAL, m18n.n('invalid_url_format'))
+
+    # Now we extract the domain and uri/path
+    domain = url[:url.index('/')]
+
+    path = url[url.index('/'):]
+    if not path.endswith('/'):
+        path = path + '/'
+
+    return domain, path
```

Comment:
A more general comment for this function is: why the hell do we merge domain+path when calling YunoHost function to split them afterward? This doesn't make any sens and make more work.

I never saw a location where we have domain+path already merged, we always merge them then split them.

I think that we should change the API and simply expect the domain and path and not their combination, this will simplify the code a lot :)